app/android/runtime_client/src/org/xwalk/app/runtime/extension/ExtensionInstanceHelper.java

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2015 Intel Corporation. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package org.xwalk.app.runtime.extension;
+
+import org.json.JSONObject;
+
+public class ExtensionInstanceHelper {
+    private BindingObjectStore mStore;
+    XWalkExtensionClient mExtension;
+    MessageHandler mHandler;
+    int mId;
+
+    public ExtensionInstanceHelper(XWalkExtensionClient extension, int id) {
+        mId = id;
+        mExtension = extension;
+        // Extend the handlers from extension object to each instance.
+        // For the case, that the developer register the handers by hand
+        // to the extenson itself.
```

Comment:
The last sentence of above comments is not very clear to me. "For the case", which case? "that" seems not needed. "register" would be "registers". "extension" is it the extension object or instance?